jax/_src/lax/control_flow/loops.py

```diff
@@ -1363,13 +1363,22 @@ def _while_partial_eval_custom(saveable, unks_in, inst_in, eqn):
   body_jaxpr_known = core.ClosedJaxpr(jaxpr_known_, body_jaxpr.consts)
   del jaxpr_known_, carry_uk_out, num_res
 
+  # Instantiate all inputs (b/c jaxpr_staged will take all inputs).
+  new_inst = [x for x, inst in zip(eqn.invars, inst_in)
+              if type(x) is core.Var and not inst]
+
   # Compute the known part of cond_fun (basically pruning inputs on known side).
   cond_unks_in = cond_consts_uk + carry_uk
   cond_jaxpr_known_, _, [cond_uk], _, _ = \
       pe.partial_eval_jaxpr_custom(
           cond_jaxpr.jaxpr, cond_unks_in, in_inst=True,
           ensure_out_unknowns=False, ensure_out_inst=True,
           saveable=ad_checkpoint.nothing_saveable)
+  # NOTE(mattjj): I think it should be impossible for cond_uk to be unknown, but
+  # asserting that led to one failure in all of Google's tests, in diffrax. So
+  # we handle it: if cond_uk is unknown, stage out the whole cond.
```

Comment:
```suggestion
  # NOTE(mattjj): I think it should be impossible for the condition to be
  # unknown, but asserting that caused a test failure in diffrax. So
  # we handle it: if it is unknown, stage out the whole cond function.
```
Was confused by "cond_uk is unknown" (and tried to maintain line count).